### PR TITLE
hw(4): More tests on if and while

### PR DESCRIPTION
**Pull request by dbernhard-0x7CD**

### CODE REVIEW

File: 04/dbernhard/tests_if7.oat [hunk 1]
```diff
@@ -0,0 +1,15 @@
+int program(int argc, string[] args) {
+    return function(0) + function(1) + function(2) + function(3);
+}
+
+int function(int x) {
+    if (x == 0) {
```
Comment (nicdard) [inline 04/dbernhard/tests_if7.oat:6]: I am not sure that this should be accepted as valid oat.

File: 04/dbernhard/tests_if8.oat [hunk 2]
```diff
@@ -0,0 +1,14 @@
+int program(int argc, string[] args) {
+    return function(0) + function(1) + function(2) + function(3);
+}
+
+int function(int x) {
+    if (x == 0) {
+
```
Comment (nicdard) [inline 04/dbernhard/tests_if8.oat:7]: Same here and in the other if branches where no return statement appears

File: 04/dbernhard/tests_if7.oat (hunk 1 above)
Comment (dbernhard-0x7CD) [reply 04/dbernhard/tests_if7.oat:6]: For the block in the if_stmt rule we choose the empty list (same as args can be empty in a fdecl)

Comment (nicdard) [reply 04/dbernhard/tests_if7.oat:6]: I meant the fact that we have an inplicit return 0 statement

File: 04/dbernhard/tests_if8.oat (hunk 2 above)
Comment (dbernhard-0x7CD) [reply 04/dbernhard/tests_if8.oat:7]: Are here actually implicit returns?

File: 04/dbernhard/tests_if7.oat (hunk 1 above)
Comment (dbernhard-0x7CD) [reply 04/dbernhard/tests_if7.oat:6]: Now I understand what you meant. I will remove this test.

File: 04/dbernhard/tests_if8.oat (hunk 2 above)
Comment (nicdard) [reply 04/dbernhard/tests_if8.oat:7]: Oh sorry I didn't see the last statement

**Review by nicdard — APPROVED** [review]

**Pull request merged**